src/Arcus.WebApi.Logging/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.WebApi.Logging
+{
+    public class ExceptionHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ExceptionHandlingMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context, ILoggerFactory loggerFactory)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception ex)
+            {
+                HandleException(context, ex, loggerFactory);
+            }
+        }
+
+        private static void HandleException(HttpContext context, Exception ex, ILoggerFactory loggerFactory)
+        {
+            var logger = loggerFactory.CreateLogger("");
```

Comment:
This looks odd, is this intentional? Is the string to reflect a "section" of the logging? Maybe use "Infrastructure instead of configurable with empty string as default?

Reply:
This is odd indeed :)
It seems that you need to provide a category for the logger or use a type parameter which will give you then a `ILogger<T>`.   Since this is a package that will be integrated in other solutions, I couldn't come up with a suitable category-name here, but I'll solve this in another way.

Reply:
I've modified it so that consumers of this package can specify the category-name when they're adding the middleware to their pipeline.
Like this:

```csharp
Func<string> getCategoryName = () => "My Category";

app.UseMiddleware<Arcus.WebApi.Logging.ExceptionHandlingMiddleware>(args: new object[] { getCategoryName });
```
or this:

```csharp
app.UseMiddleware<Arcus.WebApi.Logging.ExceptionHandlingMiddleware>(args: new object[] {"My Category" });
```

Reply:
Looks good!